pkg/structs/flamebearer/flamebearer.go

```diff
@@ -109,19 +110,47 @@ func NewProfile(name string, output *storage.GetOutput, maxNodes int) Flamebeare
 	}
 }
 
-func NewCombinedProfile(name string, output, left, right *storage.GetOutput, maxNodes int) FlamebearerProfile {
+func isEmpty(t *storage.GetOutput) bool {
+	// TODO: improve heuristic
+	return t.SampleRate == 0
+}
+
+func NewCombinedProfile(name string, left, right *storage.GetOutput, maxNodes int) (FlamebearerProfile, error) {
+	if left.Units != right.Units {
+		// if one of them is empty, it still makes sense merging the profiles
+		if left.Units != "" && right.Units != "" {
+			msg := fmt.Sprintf("left units (%s) does not match right units (%s)", left.Units, right.Units)
+			return FlamebearerProfile{}, errors.New(msg)
+		}
+	}
+
+	if left.SampleRate != right.SampleRate {
+		// if one of them is empty, it still makes sense merging the profiles
+		if left.SampleRate != 0 && right.SampleRate != 0 {
+			msg := fmt.Sprintf("left sample rate (%d) does not match right sample rate (%d)", left.SampleRate, right.SampleRate)
+			return FlamebearerProfile{}, errors.New(msg)
+		}
+	}
+
```

Comment:
@abeaumont I added some crappy validations here, anything else you find missing?

Reply:
It looks good, not sure about making the fields optional though. I think we may want to make them mandatory and report an error if any of those is missing (I _think_ they are always set up when ingesting, would have to double-check). That would increase the strictness, what do you think?

Reply:
I agree, except that there's a valid case where things are optional: when you are comparing stuff like this, where one of them is empty:
![image](https://user-images.githubusercontent.com/6951209/159919867-e4c83bea-0251-4204-80e0-a9450b079be4.png)

"why would you compare an empty one"? it may just happen, for example, your app is brand new (so left half, will be selecting an empty time range)